app/main.py

```diff
@@ -1,13 +1,16 @@
+
+
+
 def format_linter_error(error: dict) -> dict:
-    # write your code here
+    return {"line": error["line_number"], "column": error["column_number"], "message": error["text"], "name": error["code"], "source": "flake8"}
     pass
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
+    return {"errors":[format_linter_error(error) for error in errors], "path": file_path, "status": "failed" if errors else "passed"}
     pass
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
+    return [format_single_linter_file(file_path, errors) for file_path, errors in linter_report.items()]
```

Comment:
The 'pass' statement on line 16 is unnecessary because the function already has a return statement. You can remove it.